model_analyzer/reports/report_manager.py

```diff
@@ -426,6 +420,28 @@ def _find_best_configs(self, report_key):
 
         return best_run_config, best_run_config_measurement, sorted_measurements
 
+    def _create_constraint_string(self, report_key: str) -> str:
+        constraint_strs = self._build_constraint_strings()
+
+        if constraint_strs:
+            if report_key == TOP_MODELS_REPORT_KEY:
+                constraint_str = constraint_strs['default']
+            elif ',' in report_key:  # indicates multi-model
+                constraint_str = ""
+                for model_name in report_key.split(','):
+                    if model_name in constraint_strs:
+                        constraint_str += "<strong>" + model_name + "</strong>: " + constraint_strs[
+                            model_name] + "<br>&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;"
+
+                constraint_str = constraint_str[:-64]  # remove trailing <br>...
```

Comment:
what kind of wizardry is this

Reply:
![image](https://user-images.githubusercontent.com/7692737/202570236-fd787282-6e43-4340-b992-a9c11a108e02.jpg)

Reply:
Need to remove the trailing newline and spaces from the string. I couldn't think of an easier way to do this, without re-writing the loop to be aware of the terminal model.

Reply:
So you're adding endline and spaces to line up the potential next entry, and you don't know which will be the last entry, so you then delete the last one? 

Why not add the endline/spaces before instead of after, and skip the first one? Similar to how you would print a list of values with commas in between. You wouldn't add a comma after every one and then delete the last one. You would add a comma before every one EXCEPT the first one.

Reply:
I've refactored and removed the wizardry. I'm now just a muggle. Please re-review.